### PR TITLE
Better error handling

### LINKED ISSUE #678: Disconnect timeout when we can't find a VNC server
Since commit c2a4d3ef86f966580c661272d772a708f0da003d we are no longer properly handling these two cases:
- can't find a VNC server
- can't find a websocket proxy on the specified host and port combination

The proper error message is: "Failed to connect to server"

Instead we are now getting the disconnect timeout.

### PR DESCRIPTION
**Pull request by samhed**
Adds a catch fallback for uncaught errors as well as changing the way RFB._fail() works. This fixes #678 as well.

### CODE REVIEW

**Comment by samhed** [comment]
I don't think anything in here is controversial really, if someone wants to take a look you should do so soon. Otherwise I'll merge next week.

File: app/styles/base.css [hunk 1]
```diff
@@ -23,6 +23,43 @@ html {
...
+  padding: 15px;
+  width: auto;
+
+  text-align: center;
+  font-weight: bold;
+  word-wrap: break-word;
+  color: #fff;
+
+  -webkit-border-radius: 10px;
+  -moz-border-radius: 10px;
```
Comment (CendioOssman) [inline app/styles/base.css]: Not necessary with our browser requirements.

File: app/styles/base.css [hunk 2]
```diff
@@ -23,6 +23,43 @@ html {
   height:100%;
 }
 
+#fallback_error {
```
Comment (CendioOssman) [inline app/styles/base.css]: Should have a `noVNC_` prefix.

File: app/ui.js [hunk 3]
```diff
@@ -24,6 +24,21 @@ var UI;
 (function () {
     "use strict";
 
+    // Fallback for all uncought errors
+    window.onerror = function(msg, url, line) {
```
Comment (CendioOssman) [inline app/ui.js]: Should we use `addEventListener()`?

File: core/rfb.js [hunk 4]
```diff
@@ -231,13 +231,19 @@
                 this._updateConnectionState('disconnected');
                 break;
             case 'connecting':
-                this._fail('Failed to connect to server' + msg);
+                this._fail('Failed to connect to server', msg);
+                break;
+            case 'connected':
+                this._updateConnectionState('disconnecting');
+                this._updateConnectionState('disconnected');
```
Comment (CendioOssman) [inline core/rfb.js]: A comment on this sequence perhaps?

File: vnc.html [hunk 5]
```diff
@@ -65,6 +65,12 @@
 </head>
 
 <body>
+
+    <div id="fallback_error">
+        noVNC encountered an error:<br>
```
Comment (CendioOssman) [inline vnc.html]: Probably cleaner to give this it's own div instead. It's weird mixing inline and block elements.

**Comment by samhed** [comment]
@CendioOssman good points, thanks! Should all be fixed now

**Pull request merged**